salt/metalk8s/kubernetes/etcd/installed.sls

```diff
@@ -13,7 +14,15 @@ include:
 {%- set etcd_members = pillar.metalk8s.etcd.members %}
 
 {#- Compute the initial state according to the existing list of node. #}
-{%- set state = "existing" if etcd_members else "new" %}
+{%- if etcd.initial_cluster_state | string() == 'new' %}
+     {%- set state = etcd.initial_cluster_state %}
+{%- else %}
+    {%- if etcd_members is not none %}
+        {%- set state = etcd.initial_cluster_state %}
+    {%- else %}
+        {{ raise ('etcd member list cannot be empty for existing state') }}
```

Comment:
Raising here is a good idea I think, before trying to deploy anything.
You could actually move this to the for loop below:

```sls
{%- for member in etcd_members | selectattr('name') %}
  {%- do etcd_endpoints.update({member['name']: member['peer_urls'][0]}) %}
{%- else %}
  {{ raise('List of active etcd members is empty, cannot reference the existing cluster state.') }}
{%- endfor %}
```

It's a better check, because you'll get rid of not started members in the list.

Reply:
True, only that during Bootstrap, this value is actually empty and hence the raise will be called always

Reply:
Add a shortcut ;)